### ISSUE TITLE
[enterprise-4.10] Issue in file machine_management/creating-infrastructure-machinesets.adoc

### ISSUE THREAD

**Issue by mtulio** [openshift/openshift-docs#45419]
### Which section(s) is the issue in?

[Sample YAML for a machine set custom resource on Alibaba Cloud](https://docs.openshift.com/container-platform/4.10/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-alibaba_creating-infrastructure-machinesets)

### What needs fixing?

The `taints` on the MachineSet seem to be inside the `providerSpec` instead of `spec`. (Look at AWS example below on the documentation):

```diff
    spec:
      metadata:
        labels:
          node-role.kubernetes.io/infra: ""
+     taints: 
+     - key: node-role.kubernetes.io/infra
+       effect: NoSchedule
      providerSpec:
        value:
          apiVersion: machine.openshift.io/v1
...
          zoneId: <zone> 
-        taints: 
-        - key: node-role.kubernetes.io/infra
-          effect: NoSchedule
```

For reference: https://github.com/openshift/api/blob/master/machine/v1/types_alibabaprovider.go#L82

**Comment by jeana-redhat** [comment]
Thanks, Marco!

**Comment by jeana-redhat** [comment]
Merged and cherrypicked to 4.10+
